revolv/administrator/views.py

```diff
@@ -29,6 +35,423 @@ def get_context_data(self, **kwargs):
+        cashOutAdjustments = sum([payment.amount for payment in AdminAdjustment.objects.adjustments(end_date=self.start_date, cash_type='cash_out')])
+
+        cash_in = (allDonations * (1 - payment_service_fee_rate)) + (allRepayments * (1 - revolv_earnings_rate)) + cashInAdjustments
+        cash_out  = cashOutAdjustments + allReinvestments
+        net = cash_in - cash_out
+        return net
+    
+    """
+    A helper function that cleans a project name to make it JQuery 
+    friendly for use in the Accounting page.
+
+    :args:
+        name: A string to be cleaned
+
+    :return:
+        Returns a cleaned up version of name.
+    """
+    def clean_project_name(self, name):
+        name = name.replace(' ', '-').replace(';', 'sm').replace(':', 'co').replace("'", 'ap').replace('!', 'ex')
+        return name
+    
+    def make_project_dict(self):
+        self.project_dict = {}
+        for proj in Project.objects.all():
+            name = proj.title
+            self.project_dict[self.clean_project_name(name)] = name
+    
+    """
+    Updates class variables such as start_date, end_date, and project_filter
+    based on the keys passed in.
+
+    :args:
+        keys: The request passed in to the AccountingJSONView
+    """
+    def updateFilterVariables(self, keys):
+        self.make_project_dict()
+
+        try:
+            if 'start_date' in keys:
+                self.start_date = datetime.datetime.strptime(keys['start_date'], '%Y-%m-%d')
+            elif 'end_date' in keys:
+                self.end_date = datetime.datetime.strptime(keys['end_date'], '%Y-%m-%d')
+            elif 'project_choice' in keys:
+                if keys['project_choice'] == 'all':
+                    self.project_filter = Project.objects.all()
+                else:
+                    real_name = self.project_dict[keys['project_choice']]
+                    self.project_filter = Project.objects.filter(title=real_name)
+        except ValueError:
+            pass
+    
+    """
+    Sets up the basics of a JSON serializable dictionary.
+    This sets up:
+    1. cash_in and its various components (repayments, adjustments, donations)
+    2. cash_out and its various components (reinvestments, adjustments)
+    3. date_info and its various components (start_date, end_date, date_list)
+    4. cash_balances
+    5. project_names
+    6. clean_project_names
+    7. cash_in/out_adjustment_names, which is a list of all the names of adjustments created
+
+    :return:
+        Returns a properly set up dictionary.
+    """
+    def setUpFilteredDict(self):
+        filtered_dict = {'cash_in': {'repayments': {}, 'adjustments': {}, 'donations': {}}, 'cash_out': {'reinvestments': {}, 'adjustments': {}}}
+        
+        filtered_dict['date_info'] = {}
+        filtered_dict['cash_balances'] = {}
+        date_info = filtered_dict['date_info']
```

Comment:
It would be cleaner to set `date_info` to an empty dict at the beginning and then set the filled in dict to be part of the filtered_dict at the end